### PR TITLE
Revert "[DEMO for DIP1005] Converted imports to selective imports in std.array"

### PR DESCRIPTION
**Pull request by andralex**
Reverts dlang/phobos#4963

### CODE REVIEW

**Comment by andralex** [comment]
cc @CyberShadow

**Comment by CyberShadow** [comment]
OK. Wasn't trying to stir anything up, just a bit confused :)

**Comment by MartinNowak** [comment]
Test failures due to code.dlang.org not being reachable intermediately.

File: std/array.d [hunk 1]
```diff
@@ -350,7 +344,7 @@ See_Also: $(REF Tuple, std,typecons)
  */
 
 auto assocArray(Range)(Range r)
-if (isInputRange!Range)
+    if (isInputRange!Range)
```
Comment (wilzbach) [inline std/array.d]: It's an automatic revert & the spaces before the if constraint are the common Phobos style

**Comment by wilzbach** [comment]
So I rebased this to master to retrigger the CIs.

1) Imho the status before with more selective import was better and if we do such changes I would prefer if we do them automated ;-)
2) For this sole time it's safe to ignore CircleCi as it tries to checkout this branch at DMD/Druntime and fails to do so. We should probably add some fallback code to `master` if the `base_branch` does exist at the other repos.

**Comment by wilzbach** [comment]
> For this sole time it's safe to ignore CircleCi as it tries to checkout this branch at DMD/Druntime and fails to do so.

-> https://github.com/dlang/phobos/pull/4996

However
-----------

It turns out that the changing the selective import `import std.meta : AliasSeq` to a module-wide `import std.meta;` exposed leaked import symbols as e.g `std.range` as it publicly imports `std.array`.

As a consequence it was possible to use `AliasSeq` by importing `std.range` and [someone did](https://github.com/dlang/phobos/pull/4927) in the meantime - thus our CI errored correctly:

```
out/std_range_package.d(884): Error: template instance AliasSeq!(byte, ubyte, short, ushort, int, uint, long, ulong) template 'AliasSeq' is not defined
out/std_range_package.d(980): Error: template instance AliasSeq!(byte, ubyte, short, ushort, int, uint, long, ulong) template 'AliasSeq' is not defined
```

See https://github.com/dlang/phobos/pull/4981 for a fix.

File: std/array.d (hunk 1 above)
Comment (andralex) [inline std/array.d]: @wilzbach we've gone back and forth with the style. I think `if` should be flush with the previous line (no extra indent). Could you please work your magic on the autotester to make it so? :)

Comment (wilzbach) [inline std/array.d]: > we've gone back and forth with the style. I think if should be flush with the previous line (no extra indent).

https://github.com/dlang/dlang.org/pull/1534

> Could you please work your magic on the autotester to make it so?

I will give it a try :)

**Pull request merged**